Libraries/LibWeb/CSS/StyleValues/CSSMathValue.h

```diff
@@ -119,7 +119,11 @@ class CSSMathValue : public CSSStyleValue {
     bool resolves_to_number() const { return m_resolved_type.matches_number(); }
     bool resolves_to_number_percentage() const { return m_resolved_type.matches_number_percentage(); }
     Optional<double> resolve_number() const;
+    [[nodiscard]] Optional<double> resolve_number(Length::ResolutionContext const&) const;
+    Optional<double> resolve_number(Layout::Node const& layout_node) const;
     Optional<i64> resolve_integer() const;
+    [[nodiscard]] Optional<i64> resolve_integer(Length::ResolutionContext const&) const;
+    Optional<i64> resolve_integer(Layout::Node const& layout_node) const;
```

Comment:
This is an unfortunate number of overloads. 😅 (And we'll need something similar for the other `resolve_FOO()` methods too - they all can potentially have lengths in.) This API probably needs a rethink at some point.

As an actual piece of feedback: You don't need `[[nodiscard]]` here because the `Optional` class is already marked with it.